File: C Language/Assignments/Assign 1/palindrome.c

#include<stdio.h>
void main(){
    int n,original;
    int rev=0;
    printf("enter the number\n");
    scanf("%d",&n);
    original=n;
    int temp=n;
    while(temp!=0){
        int rem=temp%10;
        rev=rev*10+rem;
        temp=temp/10;
    }
    if(original==rev){
        printf("palindrome");
    }
    else{
        printf("not palindrome");
    }  
     

}

File: Java work/C to java Converted code/Function/palindrome.java
package funct;

public class palindrome {

	public static void main(String[] args) {
		int n=121;
	    
	    int k=isPalindrome(n);
	    if(k==1)
	    System.out.println("palindrome");
	    else{
	        System.out.println("not palindrome");
	    }
	}
	public static int isPalindrome(int n){
	    int rev=0;
	    int temp=n;
	    while(n>0){
	        int rem=n%10;
	        rev=rev*10+rem;
	        n=n/10;
	    }
	    if(temp==rev){
	        return 1;
	    }
	    else{
	        return 0;
	    }
		

	}

}
